### ISSUE TITLE
module load error does not set exit code

### ISSUE THREAD

**Issue by diekhans** [cea-hpc/modules#505]
### Describe the bug

when "module load" fails, it sill exists with status code zero

### To Reproduce
```console
$ module load fred
ModuleCmd_Load.c(213):ERROR:105: Unable to locate a modulefile for 'fred'
$ echo $?
0
...
```

### Expected behavior

module  load should exit with a status other than 0,  most likely 1.

Exiting with 0 on an error violates the UNIX error handling protocol and makes it impossible to use in a robust way.

**Comment by xdelaruelle** [comment]
Thanks Mark for your report.

This is important to exit with a code different than 0 in case of error. This issue was fixed in [version 4.0 of Modules](https://modules.readthedocs.io/en/latest/MIGRATING.html#non-zero-exit-code-in-case-of-error). It was even one of my first contribution to the development of this project if I remember well.

Looking at the error message you provide, it seems you still use Modules version 3.2. The development of this version stopped in 2012.

I highly recommend to upgrade to newer versions like the latest 5.3. Versions 4 and 5 are compatible with version 3.2 and fixes [the numerous bugs of this old version](https://github.com/cea-hpc/modules/milestone/1?closed=1). Moreover they provide [tons of new features](https://modules.readthedocs.io/en/latest/MIGRATING.html).

**Comment by diekhans** [comment]
thanks!!!  this is a managed system and I didn't realize they had not updated it
in eons.
